### PR TITLE
8294378: URLPermission constructor exception when using tr locale

### PR DESCRIPTION
**Pull request by ryawalla**
Hi all,

Please review this fix for  Invalid characters in hostname exception. 
This was due to the URLPermission.parseURI() method calling String.toLowerCase() and returning a non ASCII characters in the case of the tr Locale.  The fix was to use the HostPortrange.toLowerCase() method further down the stack which is Locale safe.

Thanks,
Ryan.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8294378](https://bugs.openjdk.org/browse/JDK-8294378): URLPermission constructor exception when using tr locale

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Aleksei Efimov](https://openjdk.org/census#aefimov) (@AlekseiEfimov - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10903/head:pull/10903` \
`$ git checkout pull/10903`

Update a local copy of the PR: \
`$ git checkout pull/10903` \
`$ git pull https://git.openjdk.org/jdk pull/10903/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10903`

View PR using the GUI difftool: \
`$ git pr show -t 10903`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10903.diff">https://git.openjdk.org/jdk/pull/10903.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/net/URLPermission/URLPermissionNonASCIITest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates. All rights reserved.
```
Comment (jaikiran) [inline test/jdk/java/net/URLPermission/URLPermissionNonASCIITest.java:2]: Hello Ryan, there's a comma missing here after `2022`.

**Comment by jaikiran** [comment]
I suspect it might be better to introduce that new test `@run` with Turkish locale in the existing `URLPermissionTest` itself - passing it that specific URL to test. That way you won't have to refactor that existing test to extend from a new utility class.

File: test/jdk/java/net/URLPermission/URLPermissionTest.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013, 2016, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2013, 2016, 2022, Oracle and/or its affiliates. All rights reserved.
```
Comment (jaikiran) [inline test/jdk/java/net/URLPermission/URLPermissionTest.java:2]: Hello Ryan, this should be `2013, 2022,` - we only maintain 2 years the year this file was first introduced and the latest update year.

Comment (dfuch) [reply test/jdk/java/net/URLPermission/URLPermissionTest.java:2]: Good catch!

**Review by dfuch — APPROVED** [review]
LGTM

File: test/jdk/java/net/URLPermission/URLPermissionNonASCIITest.java (hunk 1 above)
Comment (ryawalla) [reply test/jdk/java/net/URLPermission/URLPermissionNonASCIITest.java]: File has been deleted and test has been added to URLPermissionTest class

File: test/jdk/java/net/URLPermission/URLPermissionTest.java (hunk 2 above)
Comment (ryawalla) [reply test/jdk/java/net/URLPermission/URLPermissionTest.java]: updated, thanks Jaikiran

**Review by jaikiran — APPROVED** [review]
Thank you Ryan for these changes. The latest state in `e992f1f3` looks good to me.

Please run the `tier2` tests once (if not done already) before integrating.

**Review by AlekseiEfimov — APPROVED** [review]
The latest changes look good to me. Thanks for fixing it

**Comment by ryawalla** [comment]
> Thank you Ryan for these changes. The latest state in `e992f1f3` looks good to me.
> 
> Please run the `tier2` tests once (if not done already) before integrating.

Thanks Jaikiran the tests are passing

**Pull request closed without merging**